Python/marshal.c

```diff
@@ -875,17 +927,61 @@ r_long64(RFILE *p)
                                  1 /* signed */);
 }
 
+#define _w_digits(bitsize)                                              \
+static int                                                              \
+_w_digits##bitsize(uint ## bitsize ## _t *digits, Py_ssize_t size,      \
+                   Py_ssize_t marshal_ratio,                            \
+                   int shorts_in_top_digit, RFILE *p)                   \
+{                                                                       \
+    uint ## bitsize ## _t d;                                            \
```

Comment:
Just to be sure we don't call this one with invalid inputs, can you add `assert(size >= 1);` as well (even though you added it at the call site?)